### PR TITLE
add more ldap results, ldap result from type uint8 to type uint16

### PR DESCRIPTION
**Pull request by wOvAN**
More result codes taken from https://ldap.com/ldap-result-code-reference/
Result type uint8 to uint16 as the actual values exceeds uint8

### CODE REVIEW

**Comment by wOvAN** [comment]
smth wrong with golint in 1.6?  "/bin/sh: 1: golint: not found"

File: error.go [hunk 1]
```diff
@@ -8,45 +8,80 @@ import (
...
+	LDAPResultIntermediateResponse               = 113
+	LDAPResultUnknownType                        = 114
+	LDAPResultCanceled                           = 118
+	LDAPResultNoSuchOperation                    = 119
+	LDAPResultTooLate                            = 120
+	LDAPResultCannotCancel                       = 121
+	LDAPResultAssertionFailed                    = 122
+	LDAPResultAuthorizationDenied                = 123
+	LDAPResultSyncRefreshRequired                = 4096
+	LDAPResultNoOperation                        = 16654
```
Comment (liggitt) [inline error.go]: the draft-zeilenga-ldap-noop draft does not specify a value for the noOperation result code... seems odd to embed a value some implementations just picked into this library

**Comment by wOvAN** [comment]
removed nooperation, still don't get why pipeline 1.6 failes

**Comment by wOvAN** [comment]
maybe it's better to use uint32 cos winapi uses ulong (8 bytes) for result "WINLDAPAPI ULONG LDAPAPI ldap_simple_bindW( LDAP *ld, PWCHAR dn, PWCHAR passwd );" ?

**Comment by liggitt** [comment]
The ldap spec says values over ~16000 aren't allowed, so uint16 is sufficient

**Comment by vetinari** [comment]
@liggitt any objections merging this to master?

**Comment by liggitt** [comment]
I hadn't swept the new codes/messages yet. Once that's done, no objections

**Pull request merged**